### ISSUE TITLE
Drives on non-Windows systems

### ISSUE THREAD

**Issue by sburnicki** [Pash-Project/Pash#340]
The current FileSystemProvider creates the default drives from the output of `System.IO.DriveInfo::GetDrives`. Therefore we have drives like:
```
PASH> get-psdrive -PSProvider FileSystem | ft name,root
Name                                                                  Root
----                                                                  ----
/                                                                     /
/backup                                                               /backup
/boot                                                                 /boot
/run/vmblock-fuse                                                     /run/vmblock-fuse
```

While this fits what could be called a "drive" on a Linux machine, it doesn't really fit the use of a *PSDrive*. Powershell doesn't allow the characters `~ / \ . :` in a drive's name and uses the internal format of `<name>:` when it comes to recognizing a drive in a path. So we have a little conflict here.

Possible approaches:
- Allow path separators in drive names, don't make the `:` mandatory as a drive/path separator

  __Advantages:__ keep for the FS provider the native syntax

  __Disadvantages:__ we run into ambiguities, wen cannot decide on a *syntax* level what's a drive and what's the path

- Transform the drive names into a Windows-compatible form: `run_vmblock-fuse` in the example above, and simply `root:` for `/`.

  __Advantage:__ compatible to Powershell, no ambiguities

  __Disadvantages:__ not native on non-windows systems

- Only use one FS drive which is `file:`. As the rest is mounted in the root tree on Linux, everything could be accessed through that drive and the name feels native when changing/identifying the provider.

  __Advantages:__ Simple solution, will feel pretty native. Still allows custom FS drives (e.g. `home:`)

  __Disadvantages:__ We don't map the different `mounts` and devices. Maybe this could mix things up in the future, e.g. if we integrate support for real mounting on unix systems.

Does anybody have other approaches in his mind?

Currently, I'm preferring the last approach, so we don't run into error-prone ambiguities and it will still feel native.

Maybe, when supporting "mounting" and other stuff, we can add support for something like `hidden` drives. These would be the drives that I mentioned in the second approach, but they wouldn't be shown by default. Only when specifically asked for them, they can be shown and used.
This would be an extension to the normal Powershell features, which *could* cause compatibility trouble. But maybe we can also find another workaround for this when the day comes.

Any ideas about this issue?

**Comment by JayBazuzi** [comment]
The idea of mounting a device on a subdirectory is not idiomatic on Windows the way it is on *nix, but it's still possible. Your latter option (`file:`) is more appropriate.

The "PSDrive" notion is more like "/dev" on *nix, IMO.

**Comment by sburnicki** [comment]
What do you mean by `The "PSDrive" notion is more like "/dev" on *nix, IMO.`?

**Comment by JayBazuzi** [comment]
Think of how `HKLM:` is a filesystem-like presentation of part of the registry. To accomplish the same thing on *nix, you would create something in `/dev`.

**Comment by ForNeVeR** [comment]
Shouldn't `HTML` be placed somewhere in `/mnt` if it ever exist on a POSIX-compliant system?

**Comment by JayBazuzi** [comment]
Yeah, that makes sense.

**Comment by sburnicki** [comment]
Implemented in #356 as `file` drive.
